repositories/UserSocialMediaRepository.ts

```diff
@@ -0,0 +1,36 @@
+import { EntityRepository } from 'typeorm';
+import { SocialMediaType, Uuid } from 'types';
+import { UserSocialMediaModel } from '../models/UserSocialMediaModel';
+import { UserModel } from '../models/UserModel';
+import { BaseRepository } from './BaseRepository';
+
+@EntityRepository(UserSocialMediaModel)
+export class UserSocialMediaRepository extends BaseRepository<UserSocialMediaModel> {
+  public async getSocialMediasForUser(user: UserModel): Promise<UserSocialMediaModel[]> {
+    return this.getBaseFindQuery().where({ user }).getMany();
+  }
```

Comment:
I'm not sure this method needs to do the JOIN on Users. Since this would only be called if the user is already provided (e.g. in the parameter), then the joins may be redundant since we already have the user data. E.g., in `getFullUserProfile()` we are calling this method and attaching it to a `user` object, so `user.socialMedias[0].user` would be a needless cycle.